Sources/OpenCombine/Publishers/Publishers.Count.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  Publishers.Count.swift
+//
+//
+//  Created by Joseph Spadafora on 6/25/19.
+//
+
+import Foundation
+
+extension Publishers {
+
+    /// A publisher that publishes the number of elements received
+    /// from the upstream publisher.
+    public struct Count<Upstream: Publisher>: Publisher {
+
+        /// The kind of values published by this publisher.
+        public typealias Output = Int
+
+        /// The kind of errors this publisher might publish.
+        ///
+        /// Use `Never` if this `Publisher` does not publish errors.
+        public typealias Failure = Upstream.Failure
+
+        /// The publisher from which this publisher receives elements.
+        public let upstream: Upstream
+
+        public init(upstream: Upstream) {
+            self.upstream = upstream
+        }
+
+        /// This function is called to attach the specified `Subscriber`
+        /// to this `Publisher` by `subscribe(_:)`
+        ///
+        /// - SeeAlso: `subscribe(_:)`
+        /// - Parameters:
+        ///     - subscriber: The subscriber to attach to this `Publisher`.
+        ///                   once attached it can begin to receive values.
+        public func receive<SubscriberType: Subscriber>(subscriber: SubscriberType)
+            where Upstream.Failure == SubscriberType.Failure,
+                  SubscriberType.Input == Publishers.Count<Upstream>.Output
```

Comment:
Let's replace `Publishers.Count<Upstream>.Output` with just `Output` (or even `Int`, if you want).

Reply:
👍